src/storage-client/src/controller.rs

```diff
@@ -2045,21 +2136,35 @@ where
                 .expect("invalid persist usage");
 
             read.downgrade_since(&Antichain::new()).await;
-            write
-                .append(
-                    Vec::<((crate::types::sources::SourceData, ()), T, Diff)>::new(),
-                    write.upper().clone(),
-                    Antichain::new(),
-                )
-                .await
-                .expect("failed to connect")
-                .expect("failed to truncate write handle");
 
             info!(
-                "successfully truncated shard's write handle for {:?}",
+                "successfully finalized read handle for shard {:?}",
```

Comment:
Don't we also need to downgrade all the `CriticalReader`'s that were created over the lifecycle of this shard?

Reply:
Made this change; PTAL. I am not totally sure if there are other `CriticalReaderId`s or if the persist client has a way of telling me which it has outstanding.

Reply:
I don't think anyone has ever created a critical reader with any namespace other than the `CONTROLLER` one, so we are fine! But its probably a good followup someday to add the ability to expose all outstanding critical readers for a shards!